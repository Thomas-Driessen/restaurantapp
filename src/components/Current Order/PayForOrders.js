import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import ToolBar from '@material-ui/core/Toolbar';
import Container from '@material-ui/core/Container';
import { Button } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';

class PayForOrders extends React.Component {

    callStaffPay = () => {
        var tableInfo = {
            "Id": this.props.tableId,
            "PayAssistance": true
        };
        fetch('/api/table/tablePayAssistance', {
            method: 'post',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(tableInfo)
        }).then(response => response.json())
    }


    render() {
        return (
            <Grid container justify="center">
                <AppBar position="static" color="transparent" elevation={0} style={{ paddingLeft: 15, paddingRight: 20 }}>
                    <ToolBar disableGutters>
                        <Button
                            variant="contained"
                            disableElevation
                            style={{
                                float: 'left',
                                minWidth: "200px",
                                minHeight: "40px",
                                borderRadius: 50
                            }}
                            size="large"
                            color="primary"
                            onClick={this.callStaffPay}
                        >
                            Pay for Orders
                    </Button>
                        <Container disableGutters>
                            <h3 style={{ float: 'right' }}>
                                Total:  <span style={{ color: "green" }}>{(this.props.priceCurrentDrinks + this.props.priceCurrentFoods + this.props.pricePreviousDrinks + this.props.pricePreviousFoods).toFixed(2)}â‚¬</span>
                            </h3>
                        </Container>
                    </ToolBar>
                </AppBar>
            </Grid>
        )
    }
}
export default PayForOrders