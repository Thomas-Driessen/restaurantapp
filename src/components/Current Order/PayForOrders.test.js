import React from 'react';
import { render } from '@testing-library/react';
import PayForOrders from './PayForOrders';
import renderer from 'react-test-renderer';

it('PayForOrders render correctly', () => {
    const tree = renderer
        .create(<PayForOrders priceCurrentDrinks={2} priceCurrentFoods={3} pricePreviousDrinks={1} pricePreviousFoods={0} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('Render PayForOrders', () => {
    const { getByText } = render(<PayForOrders priceCurrentDrinks={2} priceCurrentFoods={3} pricePreviousDrinks={1} pricePreviousFoods={0} />);
    const linkElement = getByText(/Pay for Orders/i);
    expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
    global.gc && global.gc()
})