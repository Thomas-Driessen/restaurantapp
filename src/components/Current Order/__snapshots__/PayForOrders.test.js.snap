// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayForOrders render correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation0"
    style={
      Object {
        "paddingLeft": 15,
        "paddingRight": 20,
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "borderRadius": 50,
            "float": "left",
            "minHeight": "40px",
            "minWidth": "200px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Pay for Orders
        </span>
      </button>
      <div
        className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
      >
        <h3
          style={
            Object {
              "float": "right",
            }
          }
        >
          Total:  
          <span
            style={
              Object {
                "color": "green",
              }
            }
          >
            6.00
            â‚¬
          </span>
        </h3>
      </div>
    </div>
  </header>
</div>
`;
